### PR TITLE
Fix compile errors in GCC 5.4.0

### PR DESCRIPTION
**Pull request by kjmcqueen**
ROS-291

### CODE REVIEW

File: rosette/src/misc.cc [hunk 1]
```diff
@@ -134,13 +134,18 @@ properPrep (char* s)
 }
 #ifdef __GNUC__
 extern "C" {
-extern const char* const sys_errlist[];
-extern int sys_nerr;
+#include <string.h>
+//extern const char* const sys_errlist[];
```
Comment (pyrocto) [inline rosette/src/misc.cc]: Remove cruft.  If you need the old code at some point, git will remember it for you.

File: rosette/src/misc.cc [hunk 2]
```diff
@@ -134,13 +134,18 @@ properPrep (char* s)
...
+//extern int sys_nerr;
 };
 #endif
 
 const char*
 sys_errmsg ()
 {
-    return (errno < sys_nerr) ? sys_errlist[errno] : "unknown system error";
+ #ifdef __GNUC__
```
Comment (nashef) [inline rosette/src/misc.cc]: Remove space at beginning of line?

File: rosette/src/Ob.cc [hunk 3]
```diff
@@ -58,7 +58,11 @@ extern int nontrivial_addr_to_pre_fixnum(Ob *);
 #endif
 
 extern "C" {
+#ifdef __GNUC__
+    char* getcwd (char*);
+#else
```
Comment (nashef) [inline rosette/src/Ob.cc]: I'm dubious of this use of __GNUC__. OSX has a getcwd and surely if you use clang on Linux, then you're going to have getcwd without having __GNUC__ defined?

Comment (pyrocto) [reply rosette/src/Ob.cc]: We're just trying to get it to build correctly on Ubuntu 16.04 with gcc.  We're not going try to support building on arbitrary *nix platforms.  Greg mixed fixed-size types with platform-dependent-size types; this is preliminary to those changes.

Comment (nashef) [reply rosette/src/Ob.cc]: It's not gcc, it's g++. And, Ubuntu has both g++ and clang. Really what you're doing is trying to get the thing running using a modern C++ compiler. The default C++ standard for g++ is C++14. C++11 is a reasonable choice, at this point. But, "gnu compiler" is that reasonable.

You can check the __cplusplus macro's value to determine the C++ standard at compile time:

http://www.stroustrup.com/C++11FAQ.html#11

Comment (nashef) [reply rosette/src/Ob.cc]: FWIW, I would go for C++14, it's a good standard.

Also, at this point I think getwd is deprecated and shouldn't be used. So, probably we can just elide that code and use getcwd without any macro checks at all?

File: rosette/src/misc.cc [hunk 4]
```diff
@@ -134,13 +134,12 @@ properPrep (char* s)
 }
 #ifdef __GNUC__
 extern "C" {
-extern const char* const sys_errlist[];
-extern int sys_nerr;
+#include <string.h>
```
Comment (nashef) [inline rosette/src/misc.cc]: I think you could also include <cstring> but this is fine.

**Pull request merged**